fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -104,136 +104,155 @@ public boolean visit(String elementName, int elementIndex, Visitable visitable)
+            }
+            p.setName(searchParamCode);
+            setDateValues(p, date);
+            result.add(p);
         }
-        p.setName(searchParamCode);
-        setDateValues(p, date);
-        result.add(p);
         return false;
     }
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.DateTime dateTime) {
-        Parameter p = new Parameter();
-        if (!DATE.equals(searchParamType)) {
-            throw invalidComboException(searchParamType, dateTime);
+        if (dateTime.hasValue()) {
+            Parameter p = new Parameter();
+            if (!DATE.equals(searchParamType)) {
+                throw invalidComboException(searchParamType, dateTime);
+            }
+            p.setName(searchParamCode);
+            setDateValues(p, dateTime);
+            result.add(p);
         }
-        p.setName(searchParamCode);
-        setDateValues(p, dateTime);
-        result.add(p);
         return false;
     }
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.Decimal decimal) {
-        Parameter p = new Parameter();
-        if (!NUMBER.equals(searchParamType)) {
-            throw invalidComboException(searchParamType, decimal);
+        if (decimal.hasValue()) {
+            Parameter p = new Parameter();
+            if (!NUMBER.equals(searchParamType)) {
+                throw invalidComboException(searchParamType, decimal);
+            }
+            p.setName(searchParamCode);
+            p.setValueNumber(decimal.getValue());
+            result.add(p);
         }
-        p.setName(searchParamCode);
-        p.setValueNumber(decimal.getValue());
-        result.add(p);
         return false;
     }
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.Id id) {
-        Parameter p = new Parameter();
-        if (!TOKEN.equals(searchParamType)) {
-            throw invalidComboException(searchParamType, id);
+        if (id.hasValue()) {
+            Parameter p = new Parameter();
+            if (!TOKEN.equals(searchParamType)) {
+                throw invalidComboException(searchParamType, id);
+            }
+            p.setName(searchParamCode);
+            p.setValueCode(id.getValue());
+            result.add(p);
         }
-        p.setName(searchParamCode);
-        p.setValueCode(id.getValue());
-        result.add(p);
         return false;
     }
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.Instant instant) {
-        if (instant == null || instant.getValue() == null) {
-            return false;
-        }
-        Parameter p = new Parameter();
-        if (!DATE.equals(searchParamType)) {
-            throw invalidComboException(searchParamType, instant);
+        if (instant.hasValue()) {
```

Comment:
so, don't have to do the null check anymore?

Reply:
seems the jdbc tests passed with all the examples in the pipeline, so, I think it should be good
Nov 11, 2019 7:32:22 PM com.ibm.fhir.model.spec.test.R4ExamplesDriver processIndex
INFO: Overall success rate = 4338/4338 = 100%. Took 541359 ms

Reply:
i removed the `if (instant == null)` part to be consistent will.  per my response at https://github.com/IBM/FHIR/pull/397/files#r344880278 I don't think we need to check whether the passed Element is null, but we DO need to check that it has a nonNull value.  I favored `hasValue()` over `getValue() != null` for that.